src/parser/jobs/nin/modules/Ninki.tsx

```diff
@@ -1,178 +1,108 @@
 import {t} from '@lingui/macro'
 import {Trans, Plural} from '@lingui/react'
-import Color from 'color'
 import {ActionLink} from 'components/ui/DbLink'
-import TimeLineChart from 'components/ui/TimeLineChart'
-import {ActionKey} from 'data/ACTIONS'
 import JOBS from 'data/JOBS'
-import {CastEvent} from 'fflogs'
-import Module, {dependency, DISPLAY_MODE} from 'parser/core/Module'
-import {LegacyComboEvent} from 'parser/core/modules/Combos'
+import {Cause, Event, Events} from 'event'
+import {filter, oneOf} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
 import {Data} from 'parser/core/modules/Data'
-import {NormalisedDamageEvent} from 'parser/core/modules/NormalisedEvents'
+import {CounterGauge, Gauge as CoreGauge} from 'parser/core/modules/Gauge'
 import Suggestions, {Suggestion, TieredSuggestion, SEVERITY} from 'parser/core/modules/Suggestions'
-import React, {Fragment} from 'react'
+import React from 'react'
 
-// Constants
-const MAX_NINKI = 100
-
-const GCD_NINKI_GAIN = 5
-const FINISHER_NINKI_GAIN = 10
-const MUG_NINKI_GAIN = 40
-const MEISUI_NINKI_GAIN = 50
-const BUNSHIN_NINKI_GAIN = 5
-const SPENDER_COST = 50
-
-const ACTION_NINKI_GAIN: Array<[ActionKey, number]> = [
-	['SPINNING_EDGE', GCD_NINKI_GAIN],
-	['GUST_SLASH', GCD_NINKI_GAIN],
-	['DEATH_BLOSSOM', GCD_NINKI_GAIN],
-	['HAKKE_MUJINSATSU', GCD_NINKI_GAIN],
-	['THROWING_DAGGER', GCD_NINKI_GAIN],
-	['AEOLIAN_EDGE', FINISHER_NINKI_GAIN],
-	['ARMOR_CRUSH', FINISHER_NINKI_GAIN],
-	['SHADOW_FANG', FINISHER_NINKI_GAIN],
-	['MUG', MUG_NINKI_GAIN],
-	['MEISUI', MEISUI_NINKI_GAIN],
-]
-
-const NINKI_GCDS: ActionKey[] = [
-	'SPINNING_EDGE',
-	'DEATH_BLOSSOM',
-	'THROWING_DAGGER',
-	'SHADOW_FANG',
-]
-
-const NINKI_COMBOS: ActionKey[] = [
-	'GUST_SLASH',
-	'AEOLIAN_EDGE',
-	'ARMOR_CRUSH',
-	'HAKKE_MUJINSATSU',
-]
-
-const NINKI_OGCDS: ActionKey[] = [
-	'MUG',
-	'MEISUI',
-]
-
-const NINKI_SPENDERS: ActionKey[] = [
-	'HELLFROG_MEDIUM',
-	'BHAVACAKRA',
-	'BUNSHIN',
-]
+type GaugeModifier = Partial<Record<Event['type'], number>>
 
-interface DataPoint {
-	t: number,
-	y: number,
+// Constants
+const GCD_GAIN = 5
+const FINISHER_GAIN = 10
+const MUG_GAIN = 40
+const MEISUI_GAIN = 50
+const BUNSHIN_GAIN = 5
+const SPENDER_COST = -50
```

Comment:
It's your preference, but from the perspective of the lint gods, these constants aren't required when using the GaugeModifier Map pattern.

Reply:
I did see the other Gauge implementations were doing this without constants. I suppose I'll swap over for consistency and to cut down on line count.